File: backend/src/modules/users/user.controller.js
const userService = require('./user.service');

async function list(req, res) {
  const users = await userService.listUsers();
  res.json(users);
}

async function create(req, res) {
  const user = await userService.createUser(req.body);
  res.status(201).json(user);
}

async function listBySite(req, res) {
  const accessRows = await userService.listUsersBySite(req.params.siteId);
  res.json(accessRows);
}

module.exports = {
  list,
  create,
  listBySite,
};

File: server/src/main/java/com/legioncontrols/server/users/UserController.java
package com.legioncontrols.server.users;

import java.util.List;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

/** Java equivalent of backend/src/modules/users/user.controller.js. */
@RestController
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/api/users")
    public List<UserDto> list() {
        return userService.listUsers();
    }

    @PostMapping("/api/users")
    public ResponseEntity<UserDto> create(@RequestBody UserService.CreateUserRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(UserDto.from(userService.createUser(request)));
    }

    @GetMapping("/api/sites/{siteId}/users")
    public List<UserSiteAccessDto> listBySite(@PathVariable String siteId) {
        return userService.listUsersBySite(siteId);
    }
}
